src/rigid_pendulum_poc/solver.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+
+#include <Kokkos_Core.hpp>
+
+namespace openturbine::rigid_pendulum {
+
+/// @brief Enumerate different types of solvers
+enum class SolverType {
+    kNone = 0,
+    kDirectLinearSolver = 1,
+    kIterativeLinearSolver = 2,
+    kIterativeNonlinearSolver = 3
+};
+
+/// @brief A class for solving linear systems
+class LinearSolver {
+public:
+    LinearSolver(SolverType=SolverType::kNone);
+
+    SolverType GetSolverType() const { return solver_type_; }
+
+    Kokkos::View<double*> Solve(const Kokkos::View<double*>&, const Kokkos::View<double*>&) const;
```

Comment:
There's no need to pass `Kokkos::View` by reference - their smart-pointer like semantics mean that copies are very cheap and are actually safer with regards to object lifetime.

Reply:
Thanks for the helpful pointers on Kokkos::View here and the following comments. Clearly, I am just getting started with this container and will need to spend a little more time with it to get to the point of idiomatic use.